tests/tracer/test_settings.py

```diff
@@ -173,3 +177,29 @@ def test_x_datadog_tags(env, expected):
     with override_env(env):
         _ = Config()
         assert expected == (_._x_datadog_tags_max_length, _._x_datadog_tags_enabled)
+
+
+def test_settings_deprecation():
+    import ddtrace.settings
+
+    deprecated_names = [
+        "ConfigException",
+        "HttpConfig",
+        "Hooks",
+        "IntegrationConfig",
+    ]
+
+    assert set(deprecated_names) == set(ddtrace.settings.__all__)
+
+    for name in deprecated_names:
+        with warnings.catch_warnings(record=True) as warns:
+            warnings.simplefilter("always")
+
+            delattr(ddtrace.settings, name)
```

Comment:
Can we avoid deleting attributes in this test. We can parametrize this test or just create other test cases